src/components/Receiving/ItemDetails.js

```diff
@@ -179,11 +182,15 @@ class ItemDetails extends Component {
 
       selectedItem[key] = value;
 
+      if (value && key === 'barcode') {
+        if (!item.isSelected && value.length > 0) this.toggleItem(item);
```

Comment:
`if` nested into `if`... hmm.... could it be simplified? ... hmmm... who knows...

Reply:
you're right. I updated the condition